internal/watcher/watcher.go

```diff
@@ -57,15 +67,42 @@ func InitWatcher(filePatterns []string, callback func(), zapLogger *zap.Logger)
 }
 
 func DrainWatcher() {
-	if activeWatcher == nil {
+	if !watcherIsActive.Load() {
 		return
 	}
+	watcherIsActive.Store(false)
 	logger.Debug("stopping watcher")
 	activeWatcher.stopWatching()
 	reloadWaitGroup.Wait()
 	activeWatcher = nil
 }
 
+// TODO: how to test this?
+func retryWatching(watchPattern *watchPattern) {
+	failureMu.Lock()
+	defer failureMu.Unlock()
+	if watchPattern.failureCount >= maxFailureCount {
+		return
+	}
+	logger.Warn("watcher was closed prematurely, retrying...", zap.String("dir", watchPattern.dir))
```

Comment:
Maybe could it just be a `debug` or `info` level notice?

Reply:
I changed it to info 👍 , this should happen very rarely